website/server/models/group.js

```diff
@@ -724,6 +724,10 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
     pusher.trigger(`presence-group-${group._id}`, 'user-left', {
       userId: user._id,
     });
+
+    delete group.quest.members[user._id];
+    group.markModified('quest.members');
+    promises.push(group.save());
```

Comment:
Yeah I would agree. What about moving the creation of the update object to above, appending the update.$set value here with the new quest.members value, and having the $set portion below just append to the update.$set object?